backend/tests/unit/core/diff/test_diff_and_merge.py

```diff
@@ -96,3 +87,161 @@ async def test_diff_and_merge_schema_with_default_values(
         assert new_bool_attr.default_value is False
         new_str_attr = car_schema_main.get_attribute(name="nickname")
         assert new_str_attr.default_value == "car"
+
```

Comment:
a few new tests to make sure that the merge is correctly handling resolved conflicts